pgrx/src/callconv.rs

```diff
@@ -605,30 +651,57 @@ impl<'fcx> FcInfo<'fcx> {
         self.0
     }
 
-    /// Modifies the contained `fcinfo` struct to flag its return value as null.
+    /// Accessor for the "is null" flag
     ///
-    /// Returns a null-pointer Datum for use with the calling function's return value.
+    /// # Safety
+    /// If this flag is set to "false", then the resulting return must be a valid [`Datum`] for
+    /// the function call's result type.
+    pub unsafe fn set_return_is_null(&mut self) -> &mut bool {
+        unsafe { &mut (*self.0).isnull }
+    }
+
+    /// Modifies the function call's return to be null
     ///
-    /// Safety: If this flag is set, regardless of what your function actually returns,
-    /// Postgress will presume it is null and discard it. This means that, if you call this
+    /// Returns a null-pointer Datum for use as the calling function's return value.
+    ///
+    /// If this flag is set, regardless of what your function actually returns,
+    /// Postgres will presume it is null and discard it. This means that, if you call this
     /// method and then return a value anyway, *your extension will leak memory.* Please
-    /// ensure this method is only called for functions which, very definitely, returns null.
+    /// ensure this method is only called for functions which, very definitely, return null.
     ///
     /// # Examples
     ///
     /// ```rust,no_run
     /// use pgrx::callconv::FcInfo;
     /// use pgrx::datum::Datum;
     /// use pgrx::prelude::*;
-    /// fn foo<'a>(mut fcinfo: FcInfo<'a>) -> Datum {
-    ///     return unsafe { fcinfo.set_return_null() };
+    /// fn foo<'a>(mut fcinfo: FcInfo<'a>) -> Datum<'a> {
+    ///     unsafe { fcinfo.return_null() }
     /// }
     /// ```
     #[inline]
-    pub unsafe fn set_return_null(&mut self) -> crate::datum::Datum<'fcx> {
-        let fcinfo = unsafe { self.0.as_mut() }.unwrap();
-        fcinfo.isnull = true;
-        crate::datum::Datum::null()
+    pub fn return_null(&mut self) -> Datum<'fcx> {
```

Comment:
It is safe to return null because that means that Postgres must not examine the resulting Datum. If Postgres ignores the SQL null to examine the null Datum and dereferences it, that is Postgres being unsound. If Postgres interprets the bytes as a type that has no valid 0-initialization... how would it do that, for one...? ...but for two, that is also Postgres being unsound.